### PR TITLE
Wimpie/employee certificate controller refactor

### PR DESCRIPTION
**Pull request by Wimmas**
image

### CODE REVIEW

File: RR.App.Tests/Controllers/HRIS/EmployeeCertificationControllerUnitTests.cs [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Amqp.Transaction;
 using Moq;
 using RR.App.Controllers.HRIS;
+using RR.App.Tests.Helper;
 using RR.Tests.Data;
 using RR.Tests.Data.Models.HRIS;
+using System.Security.Claims;
 using Xunit;
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeCertificationControllerUnitTests.cs:12]: Clean: are all these necessary?

File: RR.App/Controllers/HRIS/EmployeeCertificationController.cs [hunk 2]
```diff
@@ -23,98 +24,57 @@ public EmployeeCertificationController(AuthorizeIdentity identity, IEmployeeCert
...
-        try
+        if (_identity.Role is not ("SuperAdmin" or "Admin" or "Talent" or "Journey"))
         {
-            if (_identity.Role is "SuperAdmin" or "Admin" or "Talent" or "Journey")
+            if (employeeId != _identity.EmployeeId)
             {
-                var certificates = await _employeeCertificationService.GetAllEmployeeCertifications(employeeId);
-                return Ok(certificates);
+                throw new CustomException("User data being accessed does not match user making the request.");
             }
```
Comment (bseptember) [inline RR.App/Controllers/HRIS/EmployeeCertificationController.cs:32]: Refactor: can this be reduced?

File: RR.App/Controllers/HRIS/EmployeeCertificationController.cs [hunk 3]
```diff
@@ -23,98 +24,57 @@ public EmployeeCertificationController(AuthorizeIdentity identity, IEmployeeCert
...
     {
-        try
+        if (_identity.Role is not ("SuperAdmin" or "Admin" or "Talent" or "Journey"))
         {
-            if (_identity.Role is "SuperAdmin" or "Admin" or "Talent" or "Journey")
+            if (employeeId != _identity.EmployeeId)
             {
-                var certificate = await _employeeCertificationService.GetEmployeeCertification(employeeId, certificationId);
-                return Ok(certificate);
+                throw new CustomException("User data being accessed does not match user making the request.");
```
Comment (bseptember) [inline RR.App/Controllers/HRIS/EmployeeCertificationController.cs:56]: Refactor: can this be reduced?

**Review by bseptember — CHANGES_REQUESTED** [review]

File: RR.App/Controllers/HRIS/EmployeeCertificationController.cs (hunk 2 above)
Comment (EduanO) [reply RR.App/Controllers/HRIS/EmployeeCertificationController.cs:32]:
```if (_identity.Role is not ("SuperAdmin" or "Admin" or "Talent" or "Journey") && employeeId != _identity.EmployeeId)
    throw new CustomException("Unauthorized Access.");```

File: RR.App/Controllers/HRIS/EmployeeCertificationController.cs (hunk 3 above)
Comment (EduanO) [reply RR.App/Controllers/HRIS/EmployeeCertificationController.cs:56]:
```
if (_identity.Role is not ("SuperAdmin" or "Admin" or "Talent" or "Journey") && employeeId != _identity.EmployeeId)
    throw new CustomException("Unauthorized Access.");
```

**Review by EduanO — CHANGES_REQUESTED** [review]
Please reduce the nested if statements on the controllers. I left a comment for how it can be changed.

For the unit tests, please fix the names of the methods. They have Typo's:
Employee**l**Certi**i**ficates

File: RR.App/Controllers/HRIS/EmployeeCertificationController.cs (hunk 2 above)
Comment (Wimmas) [reply RR.App/Controllers/HRIS/EmployeeCertificationController.cs]: Done

File: RR.App/Controllers/HRIS/EmployeeCertificationController.cs (hunk 3 above)
Comment (Wimmas) [reply RR.App/Controllers/HRIS/EmployeeCertificationController.cs]: Done

File: RR.App.Tests/Controllers/HRIS/EmployeeCertificationControllerUnitTests.cs (hunk 1 above)
Comment (Wimmas) [reply RR.App.Tests/Controllers/HRIS/EmployeeCertificationControllerUnitTests.cs]: Done

**Review by EduanO — APPROVED** [review]

**Review by bseptember — APPROVED** [review]

**Review by madwizzard — APPROVED** [review]

**Pull request merged**